pkg/controller/jobs/pod/pod_controller_test.go

```diff
@@ -1408,6 +1408,47 @@ func TestReconciler(t *testing.T) {
 			},
 			workloadCmpOpts: defaultWorkloadCmpOpts,
 		},
+		"if a pod group is finished and the wl is deleted, new workload shouldn't be created": {
```

Comment:
can we add another one (if it isn't there)?

If a pod group is not finished (for example, it has a non-terminated gated Pod), a workload should be recreated.